lib/Frontend/CompilerInvocation.cpp

```diff
@@ -1449,21 +1449,88 @@ static bool ParseSILArgs(SILOptions &Opts, ArgList &Args,
   // -Ounchecked might also set removal of runtime asserts (cond_fail).
   Opts.RemoveRuntimeAsserts |= Args.hasArg(OPT_RemoveRuntimeAsserts);
 
-  // If experimental move only is enabled, always enable lexical lifetime as
-  // well. Move only depends on lexical lifetimes.
-  bool enableExperimentalLexicalLifetimes =
-      Args.hasArg(OPT_enable_lexical_lifetimes) ||
-      Args.hasArg(OPT_enable_experimental_move_only);
-  // Error if both experimental lexical lifetimes and disable lexical lifetimes
-  // are both set.
-  if (enableExperimentalLexicalLifetimes &&
-      Args.hasArg(OPT_disable_lexical_lifetimes)) {
+  Optional<bool> enableLexicalBorrowScopesFlag;
+  if (Arg *A = Args.getLastArg(OPT_enable_lexical_borrow_scopes)) {
+    enableLexicalBorrowScopesFlag =
+        llvm::StringSwitch<Optional<bool>>(A->getValue())
+            .Case("true", true)
+            .Case("false", false)
+            .Default(None);
+  }
+  Optional<bool> enableLexicalLifetimesFlag;
+  if (Arg *A = Args.getLastArg(OPT_enable_lexical_lifetimes)) {
+    enableLexicalLifetimesFlag =
+        llvm::StringSwitch<Optional<bool>>(A->getValue())
+            .Case("true", true)
+            .Case("false", false)
+            .Default(None);
+  }
+  if (Args.getLastArg(OPT_enable_lexical_lifetimes_noArg)) {
+    if (!enableLexicalLifetimesFlag.getValueOr(true)) {
+      // Error if lexical lifetimes have been disabled via the meta-var form
+      // and enabled via the flag.
+      Diags.diagnose(SourceLoc(), diag::error_invalid_arg_combination,
+                     "enable-lexical-lifetimes",
+                     "enable-lexical-lifetimes=false");
+      return true;
+    } else {
+      enableLexicalLifetimesFlag = true;
+    }
+  }
```

Comment:
It appears that, for frontend options, the -enable vs. -disable form is much easier to handle.

Sorry if I indicated otherwise. I typically use `-enable-xxx=true|false` for LLVM options

No urgent need to rewrite this, unless we make have build settings start passing them...

Also for future reference. We should have some simple utility that takes two complementary boolean flags and checks that they don't conflict rather than writing custom logic

Reply:
These flags (which match https://github.com/apple/swift/pull/40473#pullrequestreview-826968897) are such that `-enable-lexical-lifetimes` has the same effect as `-enable-lexical-lifetimes=true`, so flags that are currently being passed will continue to work.